src/mem/mem_ctrl.hh

```diff
@@ -280,6 +321,21 @@ class MemCtrl : public QoS::MemCtrl
     bool retryRdReq;
     bool retryWrReq;
 
+    /**
+     * Remember if we have to retry because NVMRd queue or DRAM
+     * Fill queue were full
+     */
+    bool retryNVMRdReq;
+    bool retryNVMWrReq;
+    bool retryDRAMFillReq;
+
+    /**
+     * Current sizes of nvmReadQueue, nvmWriteQueue, dramFillQueue
+     *
+     */
+    uint64_t nvmReadQueueSize;
+    uint64_t nvmWriteQueueSize;
+    uint64_t dramFillQueueSize;
```

Comment:
I suggest making these the *max* size and just doing queue.size() for the current size. (And make them const)